<a href="https://colab.research.google.com/github/Mariajimenez16/ECO_TRAVEL_TOURS/blob/main/iaupb_simulacro_02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Etapa 1**

In [61]:
import pandas as pd

ruta = "/precipitacion_2026enero__Medellin_AeropuertoEnriqueOlayaHerrera.csv"

df = pd.read_csv(ruta, sep=';', decimal =',')

df.head()

FechaObservacion  ValorObservado
0  2026 Jan 01 01:00:00 AM             0.0
1  2026 Jan 01 01:00:00 PM             0.0
2  2026 Jan 01 01:02:00 AM             0.0
3  2026 Jan 01 01:02:00 PM             0.0
4  2026 Jan 01 01:04:00 AM             0.0

Verificar que se haya subido el .csv y todos sus datos

In [3]:
df.shape


(24102, 2)

Acá es el **punto 2**: resolver el problema de formatos mixtos de fechas

In [4]:
df["FechaObservacion"] = pd.to_datetime(
    df["FechaObservacion"],
    errors="coerce"
)

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24102 entries, 0 to 24101
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   FechaObservacion  16725 non-null  datetime64[ns]
 1   ValorObservado    24102 non-null  object        
dtypes: datetime64[ns](1), object(1)
memory usage: 376.7+ KB


**Punto 3**: Eliminar duplicados exactos (misma fecha/hora y valor)

Primero ejecutare un comando para ver cuantos duplicados tengo

In [5]:
duplicados_exactos = df.duplicated(
    subset=["FechaObservacion", "ValorObservado"]
).sum()

duplicados_exactos


np.int64(7110)

Acá eñimino los duplicados exactos, que son 7110. Por lo que usaré el .drop , ya que el drop es parecido al truncate, pero elimina completamente una info especifioca

In [25]:
df = df.drop_duplicates(
    subset=["FechaObservacion", "ValorObservado"]
)


Para comprobar que efectivamente el comando anterior funciono, usaré este otro comando y el valor al ejecutar es diferente

In [29]:
df.shape


(16428, 2)

Si funciono ya que los datos iniciales fueron: 24102 - 7110(datos duplicados) = 16992

Ahora, continuare con el **punto 4**: la eliminación de duplicados temporales (misma fecha/hora, diferente valor) dejando el de mínimo
valor.

In [10]:
df = (
    df.sort_values("ValorObservado")
      .groupby("FechaObservacion", as_index=False)
      .first()
)


Esto lo uso para verificar que si funciono y que ya no hay duplicados temporales

In [19]:
conteo_por_fecha = df["FechaObservacion"].value_counts()

fechas_duplicadas = conteo_por_fecha[conteo_por_fecha > 1]

len(fechas_duplicadas)


0

Para demostrar los resultados esperados:

Total de registros resultantes luego del procesamiento inicial, acá el resultado mermo ya que es el valor actualizado teniendi en cuenta la eliminacion de los temporales

In [32]:
total_registros_finales = len(df)

total_registros_finales


16428

Total de registros duplicados exactos y temporales eliminados

In [33]:
duplicados_eliminados_total = duplicados_exactos + 0

duplicados_eliminados_total

np.int64(7110)

Mediciones diferente de cero: mínima, máxima y más frecuente (moda)


In [62]:
minimo = valores_no_cero.min()
maximo = valores_no_cero.max()
moda = valores_no_cero.mode()

minimo, maximo, moda


(nan, nan, Series([], Name: ValorObservado, dtype: float64))

ETAPA 2